packages/garbo/src/resources/mayam.ts

```diff
@@ -0,0 +1,249 @@
+import { globalOptions } from "../config";
+
+type MayamRingSummon = {
+  choice: string;
+  value: () => number;
+  ring: number;
+};
+
+type Resonance = {
+  rings: string;
+  item: Item;
+};
+
+const RESONANCE: Resonance[] = [
+  { rings: "eye yam2 eyepatch yam4", item: $item`Mayam spinach` },
+  { rings: "vessel yam2 cheese explosion", item: $item`stuffed yam stinkbomb` },
+  { rings: "yam1 meat cheese yam4", item: $item`yam and swiss` },
+  { rings: "yam1 lightning yam3 clock", item: $item`yam battery` },
+  { rings: "yam1 yam2 yam3 explosion", item: $item`thanksgiving bomb` },
+  // {rings: "yam1 meat eyepatch yam4", item: $item`yamtility belt`}
+];
+
+const yamUses = () => get("_mayamSymbolsUsed");
+
+function findResonance(rings: string): Resonance | undefined {
+  return RESONANCE.find((resonance) => resonance.rings === rings);
+}
+
+// Function to find the resonance value
+export function findResonanceValue(input: string): number {
+  const result = findResonance(input);
+
+  // Check if the result is found
+  if (result) {
+    return garboValue(result.item);
+  }
+
+  return 0;
+}
+
+const MAYAM_RING_OPTIONS: MayamRingSummon[] = [
+  {
+    choice: "yam1",
+    value: () => garboValue($item`yam`),
+    ring: 1,
+  },
+  {
+    choice: "yam2",
+    value: () => garboValue($item`yam`),
+    ring: 2,
+  },
+  {
+    choice: "yam3",
+    value: () => garboValue($item`yam`),
+    ring: 3,
+  },
+  {
+    choice: "yam4",
+    value: () => garboValue($item`yam`),
+    ring: 4,
+  },
+  {
+    choice: "sword",
+    value: () => 0,
+    ring: 1,
+  },
+  {
+    choice: "eye",
+    value: () => 5, // +30% item drop, I don't know how to value that?
+    ring: 1,
+  },
+  {
+    choice: "chair",
+    value: () => 5, // +5 free rests, I don't know how to value that?
+    ring: 1,
+  },
+  {
+    choice: "fur",
+    value: () =>
+      Math.max(
+        0,
+        ...getExperienceFamiliars("free").map(
+          ({ expectedValue }) => expectedValue / 12,
+        ),
+      ) * 100,
+    ring: 1,
+  },
+  {
+    choice: "vessel",
+    value: () => 0, // How do we value MP regen?
+    ring: 1,
+  },
+  {
+    choice: "lightning",
+    value: () => 0,
+    ring: 2,
+  },
+  {
+    choice: "bottle",
+    value: () => 0, // How do we value HP regen?
+    ring: 2,
+  },
+  {
+    choice: "meat",
+    value: () => clamp(myLevel() * 100, 100, 1500),
+    ring: 2,
+  },
+  {
+    choice: "wood",
+    value: () => 0,
+    ring: 2,
+  },
+  {
+    choice: "wall",
+    value: () => 0,
+    ring: 3,
+  },
+  {
+    choice: "cheese",
+    value: () => garboValue($item`goat cheese`),
+    ring: 3,
+  },
+  {
+    choice: "eyepatch",
+    value: () => 0,
+    ring: 3,
+  },
+  {
+    choice: "explosion",
+    value: () => 0,
+    ring: 4,
+  },
+  {
+    choice: "clock",
+    value: () => {
+      if (globalOptions.prefs.valueOfAdventure === undefined) {
+        return 4000 * 5;
+      } else return globalOptions.prefs.valueOfAdventure * 5;
```

Comment:
Why have this fallback? Seems problematic to embed a 4000 default here.

In any case, you would write this as

```suggestion
      return (globalOptions.prefs.valueOfAdventure ?? 4000) * 5;
```

Reply:
4K is the assumed MPA for Garbo, or was? at one time?

Reply:
the valueOfAdventure preference will always be set to something, it's a requirement to run garbo

Reply:
The real answer here is we should never use `globalOptions.prefs.valueOfAdventure`--we don't actually have any reason to access `valueOfAdventure` through globalOptions, and should instead access it through `get` whenever humanly possible. It's a native mafia preference! I don't even know why it's in `globalOptions` to begin with!